espnet/nets/pytorch_backend/e2e_tts_fastspeech.py

```diff
@@ -0,0 +1,686 @@
+                                    "kaiming_uniform", "kaiming_normal"],
+                           help="How to initialize transformer parameters")
+        group.add_argument("--initial-encoder-alpha", type=float, default=1.0,
+                           help="Initial alpha value in encoder's ScaledPositionalEncoding")
+        group.add_argument("--initial-decoder-alpha", type=float, default=1.0,
+                           help="Initial alpha value in decoder's ScaledPositionalEncoding")
+        group.add_argument("--transformer-lr", default=1.0, type=float,
+                           help="Initial value of learning rate")
+        group.add_argument("--transformer-warmup-steps", default=4000, type=int,
+                           help="Optimizer warmup steps")
+        group.add_argument("--transformer-enc-dropout-rate", default=0.1, type=float,
+                           help="Dropout rate for transformer encoder except for attention")
+        group.add_argument("--transformer-enc-positional-dropout-rate", default=0.1, type=float,
+                           help="Dropout rate for transformer encoder positional encoding")
+        group.add_argument("--transformer-enc-attn-dropout-rate", default=0.1, type=float,
+                           help="Dropout rate for transformer encoder self-attention")
+        group.add_argument("--transformer-dec-dropout-rate", default=0.1, type=float,
+                           help="Dropout rate for transformer decoder except for attention and pos encoding")
+        group.add_argument("--transformer-dec-positional-dropout-rate", default=0.1, type=float,
+                           help="Dropout rate for transformer decoder positional encoding")
+        group.add_argument("--transformer-dec-attn-dropout-rate", default=0.1, type=float,
+                           help="Dropout rate for transformer decoder self-attention")
+        group.add_argument("--transformer-enc-dec-attn-dropout-rate", default=0.1, type=float,
+                           help="Dropout rate for transformer encoder-decoder attention")
+        group.add_argument("--duration-predictor-dropout-rate", default=0.1, type=float,
+                           help="Dropout rate for duration predictor")
+        group.add_argument("--init-encoder-from-teacher", default=True, type=strtobool,
+                           help="Whether to initialize encoder using teacher's parameters")
+        group.add_argument("--init-encoder-module", default="all", type=str,
+                           choices=["all", "embed"],
+                           help="Encoder modeules to be initilized")
+        # loss related
+        group.add_argument("--use-masking", default=True, type=strtobool,
+                           help="Whether to use masking in calculation of loss")
+
+        return parser
+
+    def __init__(self, idim, odim, args):
+        # initialize base classes
+        TTSInterface.__init__(self)
+        torch.nn.Module.__init__(self)
```

Comment:
## TTSInterface background

For consistency with `ASRInterface`, it is not derived from torch.nn.Module. This is because ASRInterface is available for both pytorch and chainer models. Actually, this interface is  just used for a type check at the beginning of `espnet.tts.pytorch` not to raise runtime error during training or decoding.

https://github.com/espnet/espnet/blob/5fa235912ede28f5edda10b5685ae3c88448500e/espnet/tts/pytorch_backend/tts.py#L248-L251

### torch.nn.Module

**However**, unlike ASR, TTS module can be derived from `torch.nn.Module` at this time because there is no chainer implementation. 

### AttentionTTSModel

This is OK but  I like the name `AttentionTTSInterface` because it is just interface, not implementation.

Reply:
In this PR, I keep `TTSInterface`.

Reply:
I see. I didn't notice that it is for both chainer and pytorch.  

`AttentionTTSInterface ` sounds okay for me.